### PR TITLE
[libc++] Set feature test macros __cpp_lib_ranges_contains and__cpp_lib_ranges_starts_ends_with

### PR DESCRIPTION
**Pull request by ZijunZhaoCCK**
ranges::contains: fdd089b50063
ranges::starts_with: 205175578e0d
ranges::ends_with: 0218ea4aaa54

### CODE REVIEW

File: libcxx/docs/Status/Cxx2cPapers.csv
Comment (ZijunZhaoCCK) [inline libcxx/docs/Status/Cxx2cPapers.csv:2]: @cjdb Hi Christopher, I am not sure the paper number should end with `R3` or `R4`. Both exist here: https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/  Would you mind helping review? Thank you!

Comment (H-G-Hristov) [reply libcxx/docs/Status/Cxx2cPapers.csv:2]: I think you should correct that to the latest version. Sometimes papers get new revisions.

File: libcxx/docs/ReleaseNotes/19.rst
Comment (H-G-Hristov) [inline libcxx/docs/ReleaseNotes/19.rst:43]: Maybe also add P1659R3

**Comment by H-G-Hristov** [comment]
@ZijunZhaoCCK I'd rename the title to `[libc++][ranges] Set feature test macro `__cpp_lib_ranges_contains` and `__cpp_lib_ranges_starts_ends_with` and also add links to the commits implementing the papers to the commit message (PR description above) for the record.
Otherwise it LGTM. Please wait for someone else to approve. Maybe @var-const

**Review by ldionne — APPROVED** [review]
I fixed up a few things like adding the versions to the .csv file and correcting the paper number for `ranges::contains` in the release notes. LGTM, I'll merge once the CI is green.

**Pull request merged**
